Framework/PythonInterface/mantid/geometry/src/Exports/CSGObject.cpp

```diff
@@ -5,18 +5,50 @@
 //   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 // SPDX - License - Identifier: GPL - 3.0 +
 #include "MantidGeometry/Objects/CSGObject.h"
+#include "MantidGeometry/Rendering/GeometryTriangulator.h"
+#include "MantidPythonInterface/core/Converters/NDArrayTypeIndex.h"
+#include "MantidPythonInterface/core/Converters/WrapWithNDArray.h"
 #include "MantidPythonInterface/core/GetPointer.h"
+
 #include <boost/python/class.hpp>
 #include <boost/python/copy_const_reference.hpp>
 #include <boost/python/register_ptr_to_python.hpp>
+#include <boost/python/self.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+
+#include <numpy/arrayobject.h>
+
+#define PY_ARRAY_UNIQUE_SYMBOL GEOMETRY_ARRAY_API
+#define NO_IMPORT_ARRAY
 
+using Mantid::Geometry::IObject;
+using namespace Mantid::PythonInterface::Converters;
 using Mantid::Geometry::BoundingBox;
 using Mantid::Geometry::CSGObject;
-using Mantid::Geometry::IObject;
+using Mantid::Geometry::detail::GeometryTriangulator;
 using namespace boost::python;
 
 GET_POINTER_SPECIALIZATION(CSGObject)
 
+boost::python::object wrapMeshWithNDArray(const CSGObject *self) {
+  // PyArray_SimpleNewFromData doesn't interact well with smart pointers so use raw pointer
+
+  auto localTriangulator = new GeometryTriangulator(self);
+  auto vertices = localTriangulator->getTriangleVertices();
```

Comment:
I think we can make all of these variables `const` and capture `vertices` and `triangles` by `const &` to avoid a copy?

Reply:
Tell me if I've now done this correctly or not!?

Reply:
This and the read-only flag were based on what I had added previously to Exports/MeshObject.cpp so I've made the const and ReadWrite updates to both.